plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -61,6 +64,160 @@
 _ABI = "abi"
 _BYTECODE = "bytecode"
 
+MAX_GAS_FAST = 1500
+
+# How many blocks to consider for priority fee estimation
+FEE_HISTORY_BLOCKS = 10
+
+# Which percentile of effective priority fees to include
+FEE_HISTORY_PERCENTILE = 5
+
+# Which base fee to trigger priority fee estimation at (GWEI)
+PRIORITY_FEE_ESTIMATION_TRIGGER = 100
+
+# Returned if above trigger is not met (GWEI)
+DEFAULT_PRIORITY_FEE = 3
+
+# In case something goes wrong fall back to this estimate
+FALLBACK_ESTIMATE = {
+    "max_fee_per_gas": to_wei(20,"gwei"),
+    "max_priority_fee_per_gas": to_wei(DEFAULT_PRIORITY_FEE, "gwei"),  # GWEI
+    "base_fee": None
+}
+
+PRIORITY_FEE_INCREASE_BOUNDARY = 200  # percentage
+
+
+def wei_to_gwei(number: Type[int]) -> Union[int, decimal.Decimal]:
+    """Covert WEI to GWEI"""
+    return from_wei(cast(int, number), unit="gwei")
+
+
+def round_to_whole_gwei(number: Type[int]) -> Wei:
+    """Round WEI to equivalent GWEI"""
+    gwei = wei_to_gwei(number)
+    rounded = math.ceil(gwei)
+    return to_wei(rounded, "gwei")
+
+
+def get_base_fee_multiplier(base_fee_gwei: int) -> float:
+    """Returns multiplier value."""
+
+    if base_fee_gwei <= 40:
+        return 2.0
+    elif base_fee_gwei <= 100:
+        return 1.6
+    elif base_fee_gwei <= 200:
+        return 1.4
+    else:
+        return 1.2
+
+
+def estimate_priority_fee(
+    web3: Web3,
+    base_fee_gwei: int,
+    block_number: int,
+    priority_fee_estimation_trigger: int,
+    default_priority_fee: int,
+    fee_history_blocks: int,
+    fee_history_percentile: int,
+    priority_fee_increase_boundary: int
+) -> int:
+    """Estimate priority fee from base fee."""
+
+    if base_fee_gwei < priority_fee_estimation_trigger:
+        return default_priority_fee
+
+    fee_history = web3.eth.fee_history(
+        fee_history_blocks,
+        block_number,
+        [fee_history_percentile]
+    )
+
+    rewards = sorted([reward for reward in fee_history["reward"] if reward > 0])
+    if not len(rewards):
+        return None
+
+    # Calculate percentage increases from between ordered list of fees
+    percentage_increases = [((j - i) / i) * 100 for i,
+                            j in zip(rewards[:-1], rewards[1:])]
+    highest_increase = max(*percentage_increases)
+    highest_increase_index = percentage_increases.index(highest_increase)
+
+    values = rewards.copy()
+    # If we have big increase in value, we could be considering "outliers" in our estimate
+    # Skip the low elements and take a new median
+    if highest_increase > priority_fee_increase_boundary and highest_increase_index >= len(values) // 2:
+        values = values[highest_increase_index:]
+
+    return values[len(values) // 2]
+
+
+def get_gas_price_strategy_eip1559(
```

Comment:
It would be desirable to configure a default gas_price_strategy via the LedgerApi constructor here: https://github.com/valory-xyz/open-aea/pull/15/files#diff-26c50f98b25886126548a663e3e23b7ab5f34c3599671d81e0f3efc78cf0121fR723

Including the params for the default gas_price_strategy!

Please add this functionality, including tests for it.